File: C#Part2/HomeworkStringsAndTextProcessing/ReplaceTags/ReplaceTags.cs
﻿//Write a program that replaces in a HTML document given as string all the tags <a href="…">…</a> with corresponding tags [URL=…]…/URL].

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

class ReplaceTags
{
    static void Main()
    {
        string html = "<p>Please visit <a href=\"http://academy.telerik. com\">our site</a> to choose a training course. Also visit <a href=\"www.devbg.org\">our forum</a> to discuss the courses.</p>";

        string output = Regex.Replace(html, "<a href=\"(.*?)\">(.*?)</a>", "[URL=$1]$2[/URL]");

        Console.WriteLine(output);

    }
}

File: 6 JavaScriptBasic/proverka na domashni/StringsJS/08. Replace tags/Replace tags.js
﻿// Write a JavaScript function that replaces in a HTML document given as string all
// the tags <a href="…">…</a> with corresponding tags [URL=…]…/URL].

function replace(str) {
    return str.replace(/<a href="(.*?)">(.*?)<\/a>/g, '[URL=$1]$2[/URL]')
}
var str = '<p>Please visit <a href="http://academy.telerik.com">our site</a> to choose a training course. Also visit <a href="www.devbg.org">our forum</a> to discuss the courses.</p>'
console.log(replace(str));